### PR TITLE
Adds an in-memory cache to SourcesRepository

### PR DESCRIPTION
**Pull request by florina-muntenescu**

### CODE REVIEW

File: core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt [hunk 1]
```diff
@@ -73,16 +79,22 @@ class SourcesRepository(
...
+        cache.find { it.key == source.key }?.apply { active = source.active }
         dispatchSourceChanged(source)
     }
 
     fun removeSource(source: Source) {
         dataSource.removeSource(source.key)
+        val index = cache.indexOf(source)
+        if (index >= 0) {
+            cache.removeAt(index)
+        }
```
Comment (ataulm) [inline core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt]: `cache.remove(source)` should be enough too?

File: core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt [hunk 2]
```diff
@@ -28,13 +28,18 @@ class SourcesRepository(
...
 ) {
 
+    private val cache = mutableListOf<Source>()
     private val callbacks = mutableListOf<FiltersChangedCallback>()
 
     fun registerFilterChangedCallback(callback: FiltersChangedCallback) {
         callbacks.add(callback)
     }
 
     fun getSources(): List<Source> {
```
Comment (ataulm) [inline core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt]: 1) is it ok that we're returning the cache instance directly?
2) this fun could be split into two now:

```kotlin
fun getSources(): List<Source> {
    if (cache.isEmpty()) {
        val sources = getSourcesFromSomethingBetterFunctionName()
        cache.addAll(sources)
    }
    return cache.toList()
}

private fun getSourcesFromSomethingBetterFunctionName(): List<Source> {
    // the old function
}

```

Comment (florina-muntenescu) [reply core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt]: But I am returning it as a `List<Source>`, so when it's called from the outside, you don't have access to the `MutableList` methods. You'd need to specifically cast ` repository.getSources() as MutableList` to use it as such. 
What would be the benefit of `cache.toList()`?

Comment (ataulm) [reply core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt]: there isn't much - it'll be very defensive.

Comment (florina-muntenescu) [reply core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt]: I think it's too much. As long as we're returning it as `List`, it should be ok.

**Pull request merged**